packages/xchain-bitcoincash/src/utils.ts

```diff
@@ -192,9 +192,10 @@ export const parseTransaction = (tx: Transaction): Tx => {
  */
 export const validateAddress = (address: string, network: Network): boolean => {
   try {
-    bitcash.address.toOutputScript(toLegacyAddress(address), bchNetwork(network))
-    return true
+    const toAddress = toCashAddress(address)
+    return bchaddr.isValidAddress(toAddress) && bchaddr.detectAddressNetwork(toAddress) === network
```

Comment:
Comparing `Network` (xchain) with `Network` (bchaddrjs) is currently working, but still unsafe. A simple helper might help
```ts
const toBCHAddressNetwork = (network: Network /* xchain */): bchaddr.Network => network === ' testnet' ? bchaddr.Network.Testnet : bchaddr.Network.Mainnet
```
`bchaddr.Network` is defined here https://github.com/ealmansi/bchaddrjs/blob/f3f4664b2fe5d9edba253dcc7c569b0a5e93002c/src/bchaddr.js#L39

Reply:
fixed :-)

Reply:
I was wrong to assume a type `bchaddr.Network` exists. e17339b added a `toBCHAddressNetwork` helper.